File: 프로그래머스/lv0/120583. 중복된 숫자 개수/중복된 숫자 개수.js

/*
    또 다른 방법 -> filter() 사용
    (왜 자꾸 JS의 함수들을 사용 못 하는걸까...ㅠ)

    function solution(array, n) {
        return array.filter( (e) => {
            return (e == n);
        }).length;
    }
    
*/
function solution(array, n) {
    newArr = [];
    array.forEach((e) => {
        if(e == n){
            newArr.push(n);
        }
    });
    return newArr.length;
}

File: 프로그래머스/lv0/120583. 중복된 숫자 개수/중복된 숫자 개수.java
/*
    Inhance for(햫상된 포문)을 사용해보자
    class Solution {
        public int solution(int[] array, int n) {
            int count = 0;
            for(int i : array){
                if(i == n){
                    count++;
                }
            }
            return count;
        }// solution
    }// end class
*/

class Solution {
    public int solution(int[] array, int n) {
        int count = 0;
        
        for(int i=0; i<array.length; i++){
            if(array[i] == n){
                count++;    
            }// if
        }// for

        return count;
        
    }// solution
}// end class